src/main/java/org/jenkinsci/plugins/vSphereCloudLauncher.java

```diff
@@ -153,7 +153,7 @@ public void launch(SlaveComputer slaveComputer, TaskListener taskListener)
         {
             try {
 
-                if (slaveComputer.isTemporarilyOffline()) {
+                if (slaveComputer.isOffline()) {
```

Comment:
Doc says it all
https://javadoc.jenkins.io/hudson/model/Computer.html#isTemporarilyOffline--

Reply:
According to the docs, these two are _not_ 100% equivalent: isTemporarilyOffline() says "In contrast, isOffline() represents the actual online/offline state. For example, this method may return false while isOffline() returns true if the agent failed to launch."

I'm not convinced that this code change is correct - I suspect that the code here really is more interested in whether or not there is a user-specified reason for the node being kept offline rather than it just being "accidentally" offline.
As the docs say "This method is marked as deprecated to warn people when they accidentally call this method." - the deprecation warning doesn't mean it's always wrong to call it, it just means that one has to be careful in choosing which of the two methods one means.

Personally, I would advise you to avoid changing this class unless you absolutely have to.  Unless you are 100% sure that you're fixing a bug by making this change, it's better to leave it unchanged as _any_ changes here can just as easily introduce a bug as fix one.